### PR TITLE
MAINT: Remove NPY_PY3K constant

### PR DESCRIPTION
**Pull request by sethtroisi**
After this PR NPY_PY3K should only appear in 'npy_3kcompat.h' and
'doc/Py3K.rst.txt'

I've had a lot of fun removing python2 code and wanted to say thanks to all of the wonderful maintainers who have answered my questions, promptly commented on my pulls, and provided advice along the way. numpy is amazing!
numpy_thanks

### CODE REVIEW

File: numpy/core/src/multiarray/descriptor.c [hunk 1]
```diff
@@ -1309,18 +1309,9 @@ _convert_from_type(PyObject *obj) {
     if (PyType_IsSubtype(typ, &PyGenericArrType_Type)) {
         return PyArray_DescrFromTypeObject(obj);
     }
-#if !defined(NPY_PY3K)
```
Comment (sethtroisi) [inline numpy/core/src/multiarray/descriptor.c]: This avoided being cleaned up because of merge conflicts along the way.

Comment (eric-wieser) [reply numpy/core/src/multiarray/descriptor.c]: Guilty

**Review by eric-wieser — APPROVED** [review]

**Comment by mattip** [comment]
Thanks @sethtroisi

**Comment by seberg** [comment]
Hmmm, what should happen with the stuff in `numpy/core/include/numpy/npy_3kcompat.h`?

**Comment by eric-wieser** [comment]
Is that file a public header? How do downstream libraries get their `NPY_PY3K` definition if so?

**Comment by sethtroisi** [comment]
(sorry if this is obvious) my assumption was that anyone building downstream c would include [numpy/core/include/numpy/npy_3kcompat.h:17](https://github.com/numpy/numpy/blob/master/numpy/core/include/numpy/npy_3kcompat.h#L17) which adds 
```
#ifndef NPY_PY3K
#define NPY_PY3K 1
#endif
```

But this was only an assumption.

**Pull request merged**